engine/src/main/java/org/destinationsol/game/SaveManager.java

```diff
@@ -52,17 +79,82 @@ public static void writeShip(HullConfig hull, float money, ArrayList<SolItem> it
                 }
             }
         }
-        IniReader.write(FILE_NAME, "hull", hullName, "money", (int) money, "items", sb.toString());
+
+        return sb.toString();
+
     }
 
-    public static boolean hasPrevShip() {
-        String path;
+    /**
+     * Writes the player's mercenaries to their JSON file.
+     * Will create file if it doesn't exist.
+     * @param game The instance of the game we're dealing with
+     */
+    private static void writeMercs(SolGame game) {
+        String fileName = getRsrcPath("mercenaries.json");
+        PrintWriter writer;
+        
+        ItemContainer mercIc = game.getHero().getTradeContainer().getMercs();
+        
+        List<JsonObject> jsons = new ArrayList<JsonObject>();
+        JsonObject json;
+        
+        for (List<SolItem> group : mercIc) {
+            for (SolItem item : group) {
+                SolShip merc = ((MercItem) item).getSolShip();
+                // Json fields
+                String hullName = game.getHullConfigs().getName(merc.getHull().config);
+                int money = (int) merc.getMoney();
+                
+                ArrayList<SolItem> itemsList = new ArrayList<SolItem>();
+                for (List<SolItem> group1 : merc.getItemContainer()) {
+                    for (SolItem i : group1) {
+                        itemsList.add(0, i);
+                    }
+                }
+                String items = itemsToString(itemsList);
+                
+                json = new JsonObject();
+                json.addProperty("hull", hullName);
+                json.addProperty("money", money);
+                json.addProperty("items", items);
+                
+                jsons.add(json);
+            }
+        }
+        
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        String toWrite = gson.toJson(jsons);
```

Comment:
`toWrite`
Please make it more.. sensible? At first glance, this makes me go `I'm converting something to write.. wut?`